scene/resources/packed_scene.cpp

```diff
@@ -927,95 +923,59 @@ Error SceneState::_parse_connections(Node *p_owner, Node *p_node, HashMap<String
 
 			bool exists = false;
 
 			//go through ownership chain to see if this exists
-			while (common_parent) {
+			while (common_owner) {
 				Ref<SceneState> ps;
 
-				if (common_parent == p_owner) {
-					ps = common_parent->get_scene_inherited_state();
+				if (Engine::get_singleton()->is_editor_hint()) {
+					if (common_owner == p_owner) {
+						ps = common_owner->get_scene_inherited_state();
+					} else {
+						ps = common_owner->get_scene_instance_state();
+					}
 				} else {
-					ps = common_parent->get_scene_instance_state();
+					NodeData common_owner_nd = nodes[node_map[common_owner]];
+					for (KeyValue<Variant, int> M : variant_map) {
+						if (M.value == common_owner_nd.instance) {
+							Ref<PackedScene> scene = M.key;
+							if (scene.is_valid()) {
+								ps = scene->get_state();
+							}
+						}
+					}
 				}
 
 				if (ps.is_valid()) {
-					NodePath signal_from = common_parent->get_path_to(p_node);
-					NodePath signal_to = common_parent->get_path_to(target);
+					NodePath signal_from = common_owner->get_path_to(p_node);
+					NodePath signal_to = common_owner->get_path_to(target);
 
 					if (ps->has_connection(signal_from, c.signal.get_name(), signal_to, base_callable.get_method())) {
 						exists = true;
 						break;
 					}
 				}
 
-				if (common_parent == p_owner) {
+				if (common_owner == p_owner) {
 					break;
 				} else {
-					common_parent = common_parent->get_owner();
+					common_owner = common_owner->get_owner();
 				}
 			}
 
 			if (exists) { //already exists (comes from instance or inheritance), so don't save
 				continue;
 			}
 
-			{
-				Node *nl = p_node;
-
-				bool exists2 = false;
-
-				while (nl) {
-					if (nl == p_owner) {
-						Ref<SceneState> state = nl->get_scene_inherited_state();
-						if (state.is_valid()) {
-							int from_node = state->find_node_by_path(nl->get_path_to(p_node));
-							int to_node = state->find_node_by_path(nl->get_path_to(target));
-
-							if (from_node >= 0 && to_node >= 0) {
-								//this one has state for this node, save
-								if (state->is_connection(from_node, c.signal.get_name(), to_node, base_callable.get_method())) {
-									exists2 = true;
-									break;
-								}
-							}
-						}
-
-						nl = nullptr;
-					} else {
-						if (!nl->get_scene_file_path().is_empty()) {
-							//is an instance
-							Ref<SceneState> state = nl->get_scene_instance_state();
-							if (state.is_valid()) {
-								int from_node = state->find_node_by_path(nl->get_path_to(p_node));
-								int to_node = state->find_node_by_path(nl->get_path_to(target));
-
-								if (from_node >= 0 && to_node >= 0) {
-									//this one has state for this node, save
-									if (state->is_connection(from_node, c.signal.get_name(), to_node, base_callable.get_method())) {
-										exists2 = true;
-										break;
-									}
-								}
-							}
-						}
-						nl = nl->get_owner();
-					}
-				}
-
-				if (exists2) {
-					continue;
-				}
-			}
-
```

Comment:
This code seems to have the same effect as the previous check along the ownership chain. Doesn't seem to catch some different cases.